Add support for AMD ACP_6_0 ADSP

DINESHKUMARK1 opened the pull request:
https://builds.zephyrproject.io/zephyr/pr/79796/docs/boards/amd/acp_6_0_adsp/doc/index.html

Add support for ACP_6_0 to build Sound Open Firmware with Zephyr OS on DSP.
The build target is acp_6_0. It has one Xtensa HiFi5 core, RAM & Storage: 1.75 MB HP SRAM / 512 KB IRAM/DRAM

Audio Interfaces:
1 x SP (I2S, PCM),
1 x BT (I2S, PCM),
1 x HS(I2S, PCM),
DMIC as audio interfaces.
Please check each commit message.

With these commits I have a successful build.
I obtain the zephyr.ri image which boots the DSP and loads the topology.

snd_sof_amd_acp_6_0 0000:03:00.5: booting DSP firmware
snd_sof_amd_acp_6_0 0000:03:00.5: Firmware info: version 2:11:99-29faa
snd_sof_amd_acp_6_0 0000:03:00.5: Firmware: ABI 3:29:1 Kernel ABI 3:23:0

sof@target:~$ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 2: sofacp_6_0dsp [sof-acp_6_0-dsp], device 0: I2SHS () []
Subdevices: 1/1
Subdevice #0: subdevice #0
card 2: sofacp_6_0dsp [sof-acp_6_0-dsp], device 1: I2SHS_VIRTUAL () []
Subdevices: 1/1
Subdevice #0: subdevice #0

DINESHKUMARK1 committed:
Board: amd : add board support for the Audio DSP on ACP_6_0 soc.

Create a acp_6_0_adsp board support for
the Audio DSP on ACP soc.

Signed-off-by: DineshKumar Kalva <DineshKumar.Kalva@amd.com>

DINESHKUMARK1 committed:
soc: amd: acp_6_0: add support for AMD ACP_6_0 soc.

Add a common part for AMD board ACP_6_0_ADSP.

Add support for ACP_6_0_ADSP BOARD,
which represents ACP_6_0 soc.

This has a 1 Xtensa HiFi5 core, with 200-800MHz
1.75 MB HP SRAM / 512 KB IRAM/DRAM,
1 x SP (I2S, PCM), 1 x BT (I2S, PCM), 1 x HS(I2S, PCM), DMIC as
audio interfaces.

Signed-off-by: DineshKumar Kalva <DineshKumar.Kalva@amd.com>

DINESHKUMARK1 committed:
west: sign: add support for AMD acp_6_0_adsp board.

Add support for signing acp_6_0 SOF with Zephyr images with rimage.

Signed-off-by: DineshKumar Kalva <DineshKumar.Kalva@amd.com>

DINESHKUMARK1 commented:
Due to the local repository being deleted, I was unable to reopen the original PR (79534 ). Therefore, I have raised a new PR for this issue

dcpleung commented:
Could you squash all the "update" commits to the ones originally introduced those files?

DINESHKUMARK1 commented:
> Could you squash all the "update" commits to the ones originally introduced those files?

The changes will be updated to the original commit once the CI test failure is resolved.

DINESHKUMARK1 committed:
CODEOWNERS: add codeowner for SOF with Zephyr on AMD ACP_6_0.

Add myself and basavaraj as codeowners for ACP_6_0 related files
for SOF with Zephyr OS.

Signed-off-by: DineshKumar Kalva <DineshKumar.Kalva@amd.com>

DINESHKUMARK1 commented:
> > Could you squash all the "update" commits to the ones originally introduced those files?
> 
> The changes will be updated to the original commit once the CI test failure is resolved.

Consolidate all updates into original commits.

DINESHKUMARK1 commented:
Hi there! Could you please provide an update on the status of this PR?

dcpleung commented:
Could you rearrange the commit so the SoC commit goes before the board one? It would help with git bisect. If there is a board config with a non-existant SoC config, twister would probably not be very happy about it.

Also, since there is a BSD license mixed in the files, you will need to go through the approval process.

kartben commented on boards/amd/amd_acp_6_0_adsp/board.yml:
```diff
@@ -0,0 +1,5 @@
+board:
+  name: amd_acp_6_0_adsp
```
please drop "amd_" prefix from the board name. Also needs a full name (see suggested change) 

```suggestion
  name: acp_6_0_adsp
  full_name: ACP_6_0
```

kartben replied:
full_name still missing

kartben commented on boards/amd/amd_acp_6_0_adsp/doc/index.rst:
```diff
@@ -0,0 +1,35 @@
+.. _amd_acp_6_0_adsp:
+
+AMD ACP_6_0
+############
+
```
```suggestion
.. zephyr:board:: acp_6_0

```

kartben commented on boards/amd/amd_acp_6_0_adsp/doc/index.rst:
```diff
@@ -0,0 +1,35 @@
+- Board features:
+
+  - RAM & Storage: 1.75 MB HP SRAM / 512 KB IRAM/DRAM
+  - Audio Interfaces:
+      - 1 x SP (I2S, PCM),
```
```suggestion
  - Audio Interfaces:

      - 1 x SP (I2S, PCM),
```

kartben commented on boards/amd/amd_acp_6_0_adsp/doc/index.rst:
this is really on the light side of what is typically expected for a board doc page. You may want to have a look at https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/templates/board.tmpl and in particular beef up the "Programming and Debugging" section.

DINESHKUMARK1 commented:
> https://builds.zephyrproject.io/zephyr/pr/79796/docs/boards/amd/acp_6_0_adsp/doc/index.html
> 
> Add support for ACP_6_0 to build Sound Open Firmware with Zephyr OS on DSP. The build target is acp_6_0. It has one Xtensa HiFi5 core, RAM & Storage: 1.75 MB HP SRAM / 512 KB IRAM/DRAM
> 
> Audio Interfaces: 1 x SP (I2S, PCM), 1 x BT (I2S, PCM), 1 x HS(I2S, PCM), DMIC as audio interfaces. Please check each commit message.
> 
> With these commits I have a successful build. I obtain the zephyr.ri image which boots the DSP and loads the topology.
> 
> snd_sof_amd_acp_6_0 0000:03:00.5: booting DSP firmware snd_sof_amd_acp_6_0 0000:03:00.5: Firmware info: version 2:11:99-29faa snd_sof_amd_acp_6_0 0000:03:00.5: Firmware: ABI 3:29:1 Kernel ABI 3:23:0
> 
> sof@target:~$ aplay -l **** List of PLAYBACK Hardware Devices **** card 2: sofacp_6_0dsp [sof-acp_6_0-dsp], device 0: I2SHS () [] Subdevices: 1/1 Subdevice #0: subdevice #0 card 2: sofacp_6_0dsp [sof-acp_6_0-dsp], device 1: I2SHS_VIRTUAL () [] Subdevices: 1/1 Subdevice #0: subdevice #0



the page link updated with https://builds.zephyrproject.io/zephyr/pr/79796/docs/boards/amd/acp_6_0_adsp/doc/index.html

kartben commented:
@DINESHKUMARK1  sorry for the typo in the board name
Please make your commits compliant by setting an author name/email that matches your sign-off entry

kartben commented:
@DINESHKUMARK1 I fixed the compliance check for you :) Edited the 4 commits a follows:

```
git commit --amend --author="First Last <email.address@amd.com>" --no-edit
```

DINESHKUMARK1 commented:
Could you please tell me why the CI test was cancelled?

kartben commented:
> Could you please tell me why the CI test was cancelled?

not sure -- I re-triggered it

DINESHKUMARK1 commented:
Running tests with Twister is resulting in failures. The root cause of these failures is currently unknown.

kartben commented:
> Running tests with Twister is resulting in failures. The root cause of these failures is currently unknown.

Can you try rebasing on main please? I think this is related to https://github.com/zephyrproject-rtos/zephyr/pull/79925 and should be fixed now
Also related: https://github.com/zephyrproject-rtos/zephyr/issues/79380

kartben reviewed (dismissed):
+1 for docs

kartben commented:
@DINESHKUMARK1 if the PR is not actually ready to be reviewed it might be better for you to mark it as draft. Otherwise reviewers (myself included) never know when you're actually expecting them to look at the PR to actually approve it
Also, I would really encourage you to test your documentation changes locally as it looks like doc tweaks might be one of the reasons why you're pushing new edits
See https://docs.zephyrproject.org/latest/contribute/documentation/generation.html

DINESHKUMARK1 commented:
> +1 for docs

It seems the documentation could use some additional information. Otherwise, the current details should be sufficient for the end user.

kartben commented:
> @DINESHKUMARK1 if the PR is not actually ready to be reviewed it might be better for you to mark it as draft. Otherwise reviewers (myself included) never know when you're actually expecting them to look at the PR to actually approve it Also, I would really encourage you to test your documentation changes locally as it looks like doc tweaks might be one of the reasons why you're pushing new edits See [docs.zephyrproject.org/latest/contribute/documentation/generation.html](https://docs.zephyrproject.org/latest/contribute/documentation/generation.html)

@DINESHKUMARK1 again, can you please test your changes locally, or put the PR back to draft mode so that we know it's not ready to be reviewed? Thanks!

DINESHKUMARK1 commented:
> It seems the documentation could use some additional information. Otherwise, the current details should be sufficient for the end user.

This is the final edit and is ready for review. Additional information can be added in future PRs as we have few more commits to submit. I think, the current documentation should be sufficient for this PR.

DINESHKUMARK1 commented:
> > > @DINESHKUMARK1 if the PR is not actually ready to be reviewed it might be better for you to mark it as draft. Otherwise reviewers (myself included) never know when you're actually expecting them to look at the PR to actually approve it Also, I would really encourage you to test your documentation changes locally as it looks like doc tweaks might be one of the reasons why you're pushing new edits See [docs.zephyrproject.org/latest/contribute/documentation/generation.html](https://docs.zephyrproject.org/latest/contribute/documentation/generation.html)
> > 
> > 
> > @DINESHKUMARK1 again, can you please test your changes locally, or put the PR back to draft mode so that we know it's not ready to be reviewed? Thanks!
> 
> This is the final edit and is ready for review. Additional information can be added in future PRs as we have few more commits to submit. I think, the current documentation should be sufficient for this PR.

Acknowledging PR review request, PR ready for review. Let me know if any additional info is needed.

DINESHKUMARK1 commented:
could the reviewers help with the review to progress further with next steps.

kartben commented on boards/amd/acp_6_0_adsp/doc/index.rst:
```diff
@@ -0,0 +1,77 @@
+Building
+--------
+
+Build as usual.
```
Please include actual snippet (`zephyr-app-commands`), for example to build a hello world

DINESHKUMARK1 replied:
![Screenshot 2024-11-11 193000](https://github.com/user-attachments/assets/4300b0df-6eeb-4fc2-b6a5-cbc60e46f6d7)

kartben replied:
?
This is nothing to be really helpful to someone reading the documentation in the future? I mentioned already that it would be great if you could follow the board template or look at how other boards are documentes

kartben commented on boards/amd/acp_6_0_adsp/doc/index.rst:
```diff
@@ -0,0 +1,77 @@
+
+Build as usual.
+
+For debugging purposes, use xt-gdb. For xcc debugging,
+please contact the AMD developers’ debug tools team or utilize Cadence debugging tools.
```
"please contact the AMD developers’ debug tools team" is not necessarily appropriate for the documentation of an upstream open source project IMO. Please consider dropping this.

kartben replied:
Please don't just mark items as resolved. It's up to the original reviewer to check if the comment has been correctly addressed 
If you really want to close it (e.g. if request has been applied verbatim) then please add a comment explaining what you did

kartben commented on boards/amd/acp_6_0_adsp/doc/index.rst:
```diff
@@ -0,0 +1,77 @@
+      - 1 x HS(I2S, PCM),
+      - DMIC
+
+System requirements
+*******************
+
+Xtensa Toolchain
+----------------
```
This seems to imply that the Xtensa toolchain is a requirement which I don't think is the case judging by the text that follows.

```suggestion
System requirements
*******************

Xtensa Toolchain (optional)
---------------------------
```

That being said, it does not seem possible to build with the Zephyr SDK - maybe SOF module needs updating?? https://github.com/zephyrproject-rtos/sof/tree/zephyr/src/platform
(really not familiar with this so it might be something else)


```
...
In file included from /Users/kartben/zephyrproject/zephyr/include/zephyr/arch/xtensa/arch.h:29,
                 from /Users/kartben/zephyrproject/zephyr/include/zephyr/arch/cpu.h:27,
                 from /Users/kartben/zephyrproject/zephyr/include/zephyr/kernel_includes.h:36,
                 from /Users/kartben/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from /Users/kartben/zephyrproject/zephyr/include/zephyr/pm/device.h:11,
                 from /Users/kartben/zephyrproject/zephyr/kernel/include/kernel_offsets.h:7,
                 from /Users/kartben/zephyrproject/zephyr/arch/xtensa/core/offsets/offsets.c:7:
/Users/kartben/zephyrproject/modules/hal/xtensa/include/xtensa/config/core.h:51:10: fatal error: xtensa/config/core-matmap.h: No such file or directory
   51 | #include <xtensa/config/core-matmap.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
...
```

DINESHKUMARK1 replied:
https://github.com/zephyrproject-rtos/hal_xtensa/pull/33 this PR might update SOF modules.

kartben commented on boards/amd/acp_6_0_adsp/doc/index.rst:
```diff
@@ -0,0 +1,77 @@
+For debugging purposes, use xt-gdb. For xcc debugging,
+please contact the AMD developers’ debug tools team or utilize Cadence debugging tools.
+
+Supported Features
+==================
+
+The following hardware features are supported:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| I2S       | on-chip    | interrupt controller                |
++-----------+------------+-------------------------------------+
+
+System Clock
+------------
+
+This board configuration uses a system clock frequency of @ 200 - 800MHz.
```
as mentioned before, please try to follow the board template... This should come earlier, as a subsection of the Hardware heading.

kartben replied:
This hasn't been addressed. Please check board template and other boards documentation. Also please note how comment being made usually clearly indicate which Line range they apply to -- here, for example, it's not just the system clock section I was referring to... (and as you can see this is a very good example of why one should mark conversations as resolved themselves :))

DINESHKUMARK1 replied:
Currently Zephyr is being enabled on AMD reference/proprietary boards. So, we couldn't add more data as part of boards documentation. We don't have any generic boards or evaluation boards to provide more information currently. We will plan to add more information as and when possible for everyone's usage. We have just done the basic enablement of Zephyr, still we have many changes to add, so we will plan to update the documentation

kartben commented on boards/amd/acp_6_0_adsp/doc/index.rst:
```diff
@@ -0,0 +1,77 @@
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| I2S       | on-chip    | interrupt controller                |
```
I am not sure what you mean here? Is I2S actually supported? and why "interrupt controller"?

kartben commented on boards/amd/acp_6_0_adsp/board.yml:
you also need an ``acp_6_0_adsp.yml`` file so that the board is tested in CI (which would also allow to catch the apparent issue of it not currently working with the Zephyr SDK)

kartben reviewed (changes requested):
> could the reviewers help with the review to progress further with next steps.

I made a comment a couple weeks ago about the documentation being on the light side, and it is now even lighter, so please try to improve it a bit, I left some comments. Also, please look at making sure the board can be used with Zephyr SDK and is tested/testable in CI
Thanks

henrikbrixandersen commented:
> could the reviewers help with the review to progress further with next steps.

We are currently in feature freeze for Zephyr v4.0.0, so most developers are focusing on testing, bug fixes, documentation updates etc. at the moment. I am sure reviews will be ticking in once v4.0.0 is released.

andyross commented on soc/amd/acp_6_0/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+
+if SOC_ACP_6_0
+config DCACHE_LINE_SIZE
+default 128
```
Style: lines under a "config" should be indented with a single tab.

andyross commented on boards/amd/acp_6_0_adsp/acp_6_0_adsp.yml:
```diff
@@ -0,0 +1,14 @@
+arch: xtensa
+toolchain:
+  - zephyr
+  - xcc
```
FWIW, pretty much every other SOF platform is moving towards xt-clang at this point.  The last Cadence toolchain version that supports xcc is now over three years old.

andyross commented on boards/amd/acp_6_0_adsp/acp_6_0_adsp_defconfig:
```diff
@@ -0,0 +1,12 @@
+CONFIG_OUTPUT_SYMBOLS=y
+CONFIG_MULTI_LEVEL_INTERRUPTS=n
+CONFIG_2ND_LEVEL_INTERRUPTS=n
+CONFIG_BUILD_OUTPUT_BIN=n
```
This one should be forced off at the arch layer, no?  If you're getting a zephyr.bin (which is mostly only meaningful on cortex-M) there's a bug somewhere we should fix.   It's not supposed to be something every board needs to turn off.

andyross commented on boards/amd/acp_6_0_adsp/acp_6_0_adsp_defconfig:
```diff
@@ -0,0 +1,12 @@
+CONFIG_MULTI_LEVEL_INTERRUPTS=n
+CONFIG_2ND_LEVEL_INTERRUPTS=n
+CONFIG_BUILD_OUTPUT_BIN=n
+CONFIG_CLEANUP_INTERMEDIATE_FILES=y
```
This one should come out.  The intermediate files are actually useful for a lot of things.  This setting exists for e.g. CI applications that need to have hundreds or thousands of build trees in cloud filesystems with limited storage.  Generic builds should use the defaults, it's not something the board layer is supposed to care about.

andyross reviewed (commented):
Sorry, I missed this when it came by the first time.  Notes and nitpicks, nothing particularly fatal.

andyross reviewed (approved):
I'm out of complaints
